clarsach/spectrum.py

```diff
@@ -12,8 +12,56 @@
 
 __all__ = ['XSpectrum']
 
+def calculate_flux_spectrum(spec, counts=None):
+    """
+    Convert a spectrum from units of photon counts to
+    units of photon flux.
+
+    **Warning**: You should do this for plotting *only*.
+    Converting from flux to counts, as is done in the detector,
+    involves a convolution with the response matrix of the detector.
+    Deconvolving this correctly is very hard, and not done in this
+    function. This is for visualization purposes only!
+
+    Parameters
+    ----------
+    spec : clarsach.XSpectrum object
+        An XSpectrum object containing the X-ray spectrum
+
+    counts : iterable
+        An array with counts to be converted to the flux
+        If this is None, use the `counts` attribute of the `spec`
+        object.
+
+    Returns
+    -------
+    flux_spectrum:
+        The flux spectrum
+
+    """
+
+    # make a flat spectrum so that I can integrate
+    # ARF and RMF only
+    flat_model = np.ones_like(spec.counts) * (spec.rmf.energ_hi -
```

Comment:
Well, so I'm pretty sure it's the unfolding, not the application of the ARF/RMF, because I tested those on simulated spectra with a direct comparison to sherpa, and that gives me correct results for all instruments I tested so far. 

Here's a slide I found in a [etalk by David Huenemoerder](http://cxc.cfa.harvard.edu/xrayschool/talks/intro_xray_analysis.pdf) that deals with this: 
<img width="1038" alt="screen shot 2017-07-14 at 17 30 25" src="https://user-images.githubusercontent.com/5237391/28231570-36bd93d0-68ba-11e7-9c0f-8ff220eefdf7.png">

I find that incredibly confusing, though. The words sound like we should be integrating: if S(E) = 1, then we want the counts per unit flux, then we should be integrating S(E) in order to get something that's in the same units as the data. But the equation makes it look like we shouldn't be integrating. He does mention, though, that you can find spurious features. I've been trying to figure out how exactly sherpa computes this plot, but so far, I've not managed to track it down within the source code.

Do you think it would help if we talked this through per telecon rather that GitHub issues next week?